Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/orientation.h

```diff
@@ -334,6 +350,91 @@ void reverse_face_orientations(const FaceRange& face_range, PolygonMesh& pmesh)
     }
 }
 
+/**
+* \ingroup PMP_orientation_grp
+* makes the orientation of all the connected_components of a `TriangleMesh` positive.
+* A closed polygon mesh is considered to have a positive orientation if the normal vectors
+* to all its faces point outside the domain bounded by the polygon mesh.
+* The normal vector to each face is chosen pointing on the side of the face
+* where its sequence of vertices is seen counterclockwise.
+*
+* @tparam TriangleMesh a model of `FaceListGraph` and `MutableFaceGraph` .
+* @tparam NamedParameters a sequence of \ref namedparameters
+*
+* @param tm a triangulated `TriangleMesh`
+* @param np optional sequence of \ref namedparameters among the ones listed below
+*
+* \cgalNamedParamsBegin
+*   \cgalParamBegin{vertex_point_map}
+*     the property map with the points associated to the vertices of `tm`.
+*     If this parameter is omitted, an internal property map for
+*     `CGAL::vertex_point_t` should be available in `TriangleMesh`
+*   \cgalParamEnd
+*   \cgalParamBegin{face_index_map}
+*     a property map containing the index of each face of `tm`.
+*   \cgalParamEnd
+* \cgalNamedParamsEnd
+*/
+template<class TriangleMesh, class NamedParameters>
+void orient_connected_components(TriangleMesh& tm, const NamedParameters& np)
+{
+  typedef boost::graph_traits<TriangleMesh> GT;
+  typedef typename GT::vertex_descriptor vertex_descriptor;
+  typedef typename GT::face_descriptor face_descriptor;
+  typedef typename GetVertexPointMap<TriangleMesh,
+                                     NamedParameters>::const_type Vpm;
+  typedef typename GetFaceIndexMap<TriangleMesh,
+                                   NamedParameters>::const_type Fid_map;
+
+  if (!is_triangle_mesh(tm)) return ;
+
+  Vpm vpm = boost::choose_param(get_param(np, internal_np::vertex_point),
+                                get_const_property_map(boost::vertex_point, tm));
+
+  Fid_map fid_map = boost::choose_param(get_param(np, internal_np::face_index),
+                                        get_const_property_map(boost::face_index, tm));
+
+  std::vector<std::size_t> face_cc(num_faces(tm), std::size_t(-1));
+
+  // set the connected component id of each face
+  std::size_t nb_cc = connected_components(tm,
+                                bind_property_maps(fid_map,make_property_map(face_cc)),
+                                parameters::face_index_map(fid_map));
+
+  if (nb_cc == 1)
```

Comment:
if `nb_cc = 1`, the only connected component should be oriented anyhow, right?

Reply:
It depends. One of the things I wanted to see with Sebastien is : Should the function orient everything positively, in which case you are right, or should it orient everything in the direcetion of the first CC (which seems less useful to me), in which case this test should remain here.

Reply:
Ok I understand. But the doc above says the function orients everything positively, and it is also what the code below does.
plus - you can always use `reverse_face_orientations()` to change the orientation of everything a posteriori. The point is to have all the connected components consistently (more or less) no matter on which side.
But let's wait for Sebastien's answer.
cc @sloriot

Reply:
The only thing that is important is that considering the nesting of meshing they are consistently oriented. The orientation of the outer shells is not important. Maybe the function should have a parameter to indicate whether the outer shells must be inward or outward oriented (if the mesh is rather large we might want to avoid to flip the orientation of all faces).